internal/provider/resource_rediscloud_subscription.go

```diff
@@ -241,6 +241,7 @@ func resourceRedisCloudSubscription() *schema.Resource {
 							Description: "A meaningful name to identify the database",
 							Type:        schema.TypeString,
 							Required:    true,
+							ValidateDiagFunc: validateDiagFunc(validation.StringLenBetween(0, 40)),
```

Comment:
Might be worth using the SDK's [`validateDiagFunc`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.4/helper/validation#ToDiagFunc) over this custom one? Unless I've missed some extra functionality it has!
```suggestion
							ValidateDiagFunc: validation.ToDiagFunc(validation.StringLenBetween(0, 40)),
```